bin/faucet/src/client.rs

```diff
@@ -54,32 +52,64 @@ pub struct FaucetClient {
     rng: RpoRandomCoin,
 }
 
+// TODO: Remove this once https://github.com/0xPolygonMiden/miden-base/issues/909 is resolved
 unsafe impl Send for FaucetClient {}
-unsafe impl Sync for FaucetClient {}
 
 impl FaucetClient {
-    pub async fn new(config: FaucetConfig) -> Result<Self, InitError> {
-        let (rpc_api, root_block_header, root_chain_mmr) =
-            initialize_faucet_client(config.clone()).await?;
-
-        let (faucet_account, account_seed, secret) = build_account(config.clone())?;
-        let faucet_account = faucet_account;
-        let id = faucet_account.id();
+    /// Creates a new faucet client.
+    pub async fn new(config: &FaucetConfig) -> Result<Self, ClientError> {
+        let (mut rpc_api, root_block_header, root_chain_mmr) =
+            initialize_faucet_client(config).await?;
+
+        let faucet_account_data = AccountData::read(&config.faucet_account_path)
+            .context("Failed to load faucet account from file")?;
+
+        let id = faucet_account_data.account.id();
+
+        info!(target: COMPONENT, "Requesting account state from the node...");
+        let faucet_account = match request_account_state(&mut rpc_api, id).await {
+            Ok(account) => {
+                info!(
+                    target: COMPONENT,
+                    hash = %account.hash(),
+                    nonce = %account.nonce(),
+                    "Received faucet account state from the node",
+                );
+
+                account
+            },
+
+            Err(err) => match err {
+                ClientError::RequestError(status) if status.code() == tonic::Code::NotFound => {
+                    info!(target: COMPONENT, "Faucet account not found in the node");
+
+                    faucet_account_data.account
+                },
```

Comment:
Is the idea that the mint transaction will generate the account in this scenario?

If so, could we not move that into the `CreateFaucetAccount` command? And then fail on all errors here instead.

Reply:
> It was my first try, but then I realized that it's not a straightforward thing. In order to submit transaction, we need to instantiate client with executor, storage and so on, so it will require at least refactoring of the client and all the stuff it uses for the transaction execution and submission. I'm not sure this is worth such refactoring, but I'm open to discussion.

That's a fair point. I think overall the faucet code could do with a larger refactoring but definitely as a separate issue when we have some extra bandwidth again.

> The second challenging thing for me here is that we would need to create faucet account separately from minting transaction and I'm not sure if I understand how to do it correctly. :) Will it be enough to just omit execution script from transaction args and keep the remaining the same?

Good question. I have no idea :D